libexec/bats-core/bats

```diff
@@ -60,7 +60,7 @@ expand_path() {
   printf -v "$result" '%s/%s' "$dirname" "${path##*/}"
 }
 
-BATS_LIBEXEC="$(dirname "$(expand_link "$BASH_SOURCE")")"
+BATS_LIBEXEC="$(dirname "$(expand_link "${BASH_SOURCE[0]}")")"
```

Comment:
https://github.com/koalaman/shellcheck/wiki/SC2128

This was using `BASH_SOURCE` only for the first element, so it should be safe to fix it this way.